app/services/candidate_submission.rb

```diff
@@ -0,0 +1,71 @@
+class CandidateSubmission
+  include Candidate
+
+  def initialize(session, step_name)
+    @session = session
+    @step_name = step_name
+  end
+
+  def call
+    sign_up_candidate(select_candidate)
+  end
+
+private
+
+  def sign_up_candidate(body)
+    ApiClient.sign_up_teacher_training_adviser_candidate(body)
+  end
+
+  def select_candidate
+    case @step_name
+    when "accept_privacy_policy"
+      filter_returner_candidate
+    when "degree/accept_privacy_policy"
+      filter_degree_candidate
+    when "studying/accept_privacy_policy"
+      filter_studying_candidate
+    when "equivalent/accept_privacy_policy"
+      filter_equivalent_candidate
+    end
+  end
+
+  def filter_returner_candidate
```

Comment:
These methods are rinse and repeat; can we DRY it up by passing in the constant, so `DEGREE/RETURNER/etc` to a single method to filter the session based on the candidate type?

Reply:
Good idea